### PR TITLE
Polish editor PostFormat and PostFormatPanel components.

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
This PR refactors two components in the `editor` package: `PostFormat` and `PostFormatPanel`. The changes made are designed to simplify the code and make it easier to implement changes I want to make in #24024.

### List of changes
- I reduced Lodash usage, preferring more explicit defaulting for `undefined`/`null` handling using nullish coalescing and native JS array methods.
- I optimized `PostFormatPanel`'s `useSelect` callback to return only a single value.
- I optimized `PostFormatPanel`'s `PostFormatSuggestion` to take a single argument for the callback, rather than one for a callback and one for a value to pass to that callback.
- I simplified `PostFormatPanel`'s contained `PanelBody` to use a `Fragment` for its title element, rather than an array of elements with `key` props.
- I simplifed both `PostFormat` and `PostFormatPanel` to move a lot of the logic into the `useSelect` callbacks, to avoid unnecessarily running the same checks on every render.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/editor/src/components/post-format/index.js [hunk 1]
```diff
@@ -55,57 +79,30 @@ function PostFormat( {
...
-							value: format.id,
+						options={ listedFormats.map( ( format ) => ( {
+							label: POST_FORMAT_TITLES[ format ],
+							value: format,
 						} ) ) }
 					/>
 				</div>
 
-				{ suggestion && suggestion.id !== postFormat && (
+				{ suggestedFormat && (
```
Comment (talldan) [inline packages/editor/src/components/post-format/index.js:92]: Similar to the previous comment, the code changes `suggestedFormat` to be validated against `supportedFormats` which logically causes an issue when that values doesn't exist in `POST_FORMAT_TITLES`.

Previously the value was sourced from `POST_FORMATS`.

**Review by talldan — CHANGES_REQUESTED** [review]
I've reviewed half of this, but as a reviewer I've found this refactor has too many logic changes and parts that won't work the same as before.

A good rule of thumb for a refactor is that it should unit test the same as it did previously (if it had tests).

How about an intermediate PR where we refactor the existing component to hooks without changing logic?

**Comment by ZebulanStanphill** [comment]
@talldan
> How about an intermediate PR where we refactor the existing component to hooks without changing logic?

Good idea. I'll start splitting out smaller PRs from this one, starting with #26238.

**Comment by talldan** [comment]
Thanks for doing that. Merged that one just now. What's the next step?

File: packages/editor/src/components/post-format/index.js [hunk 2]
```diff
@@ -1,51 +1,75 @@
...
-	{ id: 'link', caption: __( 'Link' ) },
-	{ id: 'image', caption: __( 'Image' ) },
-	{ id: 'quote', caption: __( 'Quote' ) },
-	{ id: 'standard', caption: __( 'Standard' ) },
-	{ id: 'status', caption: __( 'Status' ) },
-	{ id: 'video', caption: __( 'Video' ) },
-	{ id: 'audio', caption: __( 'Audio' ) },
-	{ id: 'chat', caption: __( 'Chat' ) },
-];
+export const POST_FORMAT_TITLES = {
```
[the remark this answers is not in the document]
Comment (ZebulanStanphill) [reply packages/editor/src/components/post-format/index.js:19]: Thanks for pointing this out. I'm going to switch back to the array.

Your comment also made me realize that the post formats, even as an array, aren't sorted at all, so I created #26305 to fix that.

**Comment by ZebulanStanphill** [comment]
@talldan Alright, now that #26305 is merged, I've rebased this PR, and it should be ready for review again. It now contains only code style and optimization tweaks, and should be much easier to review than before.

**Comment by ZebulanStanphill** [comment]
Ultimately, this PR is basically just nitpicking, and most of the more useful changes have made it into `trunk` via other PRs, so I'm closing this one, since I lack the spare time to spend on it.

**Pull request closed without merging**